neofetch

```diff
@@ -8823,6 +8833,82 @@ EOF
     fi
 }
 
+ssh_connect(){
+    ssh_command="ssh"
+    ssh_opts="-t"
+
+    # filter out first --ssh HOST argument before passing to remote host
+    ssh_arg=no
+    newargs=""
+    while [[ "$1" ]]
+    do
+        if [[ "$ssh_arg" == "no" ]] && [[ "$1" == "--ssh" ]]
+        then
+            shift 2
+            ssh_arg=yes
+            continue
+        fi
+        newargs="$newargs $( printf "%q" "$1" )"
+        shift
+    done
+
+    # generate random filename (with fallbacks)
+    if type -p uuidgen > /dev/null; then
+        randname=$( uuidgen )
+    elif type -p xxd > /dev/null; then
+        randname=$( head -c 20 /dev/urandom | xxd -ps )
+    elif type -p openssl > /dev/null; then
+        randname=$( openssl rand -hex 20 )
+    else
+        randname="$RANDOM-$RANDOM-$RANDOM-$RANDOM-$RANDOM-$RANDOM"
+    fi
+    REMOTE_SCRIPT="/tmp/neofetch-$randname.sh"
+
+    # pipe commands to ssh
+    (
+        # workaround for old systems not providing base64 command directly
+        base64enc(){
```

Comment:
Why does the script need to be run through `base64` at all?

Reply:
To prevent variable substitution, but this can actually by achieved with noninterpolating heredoc.